utils/s2n_safety.h

```diff
@@ -81,6 +81,14 @@ static inline void* trace_memcpy_check(void *restrict to, const void *restrict f
 #define GUARD( x )      if ( (x) < 0 ) return -1
 #define GUARD_PTR( x )  if ( (x) < 0 ) return NULL
 
+/* TODO: use the OSSL error code in error reporting https://github.com/awslabs/s2n/issues/705 */
+#define GUARD_OSSL( x , errcode )			\
+  do {							\
+  if (( x ) == 0) {					\
```

Comment:
Does s2n build with both `== 0` and `!= 1`? I was leaning towards `!= 1`, but not enough to block the PR. Going with `!= 1` would cause s2n to exit with an error if OpenSSL broke it's API contract, which seems better than continuing as normal. 

How do others feel? Any likely performance benefit for one or the other?

Reply:
There is probably a small performance benefit for the ```== 0``` test, since on many RISC architectures comparing against 0 is easier than comparing against a non-zero constant.  But its likely to be a very small advantage.  On CISC such as x86 it makes no difference:

```
	.cfi_def_cfa_register %rbp
	subq	$16, %rsp
	movl	$1, %esi
	movl	%edi, -8(%rbp)
	movl	-8(%rbp), %edi
	callq	_add
	cmpl	$0, %eax
	jne	LBB1_2
```

```
	.cfi_def_cfa_register %rbp
	subq	$16, %rsp
	movl	$1, %esi
	movl	%edi, -8(%rbp)
	movl	-8(%rbp), %edi
	callq	_add
	cmpl	$1, %eax
	je	LBB2_2
```

So we might as well compare against ```!= 1```